src/Assessor/Filter/VotePlaceFilters.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace AppBundle\Assessor\Filter;
+
+use AppBundle\Exception\ProcurationException;
+use Doctrine\ORM\QueryBuilder;
+use Symfony\Component\HttpFoundation\Request;
+
+class VotePlaceFilters extends AssessorFilters
+{
+    public const UNASSOCIATED = 'unassociated';
+    public const ASSOCIATED = 'associated';
+
+    public static function fromQueryString(Request $request)
+    {
+        $filters = parent::fromQueryString($request);
+        $filters->setStatus($request->query->get(self::PARAMETER_STATUS, self::UNASSOCIATED));
+
+        return $filters;
+    }
+
+    public function setStatus(string $status): void
+    {
+        $status = mb_strtolower(trim($status));
+
+        if (!\in_array($status, [self::ASSOCIATED, self::UNASSOCIATED])) {
+            throw new ProcurationException(sprintf('Unexpected vote place status "%s".', $status));
+        }
+
+        parent::setStatus($status);
+    }
+
+    public function apply(QueryBuilder $qb, string $alias): void
+    {
+        parent::apply($qb, $alias);
+
+        $status = $this->getStatus();
+
+        if (self::UNASSOCIATED === $status) {
+            $qb
+                ->andWhere(
+                    $qb->expr()->orX(
+                        $qb->expr()->andX(
```

Comment:
hmm, 

```
(A = 1 AND B = 0)
OR
(A = 0 AND B = 1)
OR
(A = 1 AND B = 1)
```
for me it's equal to: `A = 1 OR B = 1`, so you can replace your ORs blocks with this:
```
->andWhere($qb->expr()->orX(
  $alias.'.substitudeOfficeAvailable = true',
  $alias.'.holderOfficeAvailable = true'
))
```
WDYT?